app/controllers/topics_controller.rb

```diff
@@ -16,6 +16,14 @@ def create
     end
   end
 
+  def show
+    @topic = Topic.find_by_id(params[:id])
+
+    if @topic.nil?
```

Comment:
This is a nice thought, but I never see this kind of care in professional projects and you'll quickly become exhausted putting this in for every find. I'd say remove this if statement.

Generally, we rely on the http status codes which rails automatically handles for us. When something is not found a 404 not found error is raised and the browser requests the apps 404 page (which is kept in the public folder), so if you want a nice message for when something is not found, just style your 404 page and put a link to the home page in it 

In addition to the 404 page there's also the 422 (unprocessable) and 500 (something unexpected) pages.

Reply:
Also, whenever you have an if statement (or some other logical condition), you need to test it somewhere. This kind of thing loses you points on test coverage reports.